src/ConstructiveSolidGeometry/PointsAndVectors/Points.jl

```diff
@@ -238,17 +238,34 @@ struct CylindricalPoint{T} <: AbstractCylindricalPoint{T}
     CylindricalPoint{T}(r::Real, φ::Real, z::Real) where {T} = new(T(r), mod(T(φ),T(2π)), T(z))
 end
 
-# Handles Unitful quantities or plain numbers 
 function CylindricalPoint(r, φ, z)
-    r_val = isa(r, Unitful.Length) ? float(to_internal_units(r)) : float(r)
-    φ_val = isa(φ, Unitful.Quantity) ? float(to_internal_units(φ)) : float(φ)
-    z_val = isa(z, Unitful.Length) ? float(to_internal_units(z)) : float(z)
+    if !(r isa Real || r isa Unitful.Length)
+        throw(ArgumentError("Expected `r` to be a length or Real, got unit $(Unitful.unit(r))"))
+    end
 
-    T = promote_type(typeof(r_val), typeof(φ_val), typeof(z_val))
-    CylindricalPoint{T}(T(r_val), T(φ_val), T(z_val))
+    if !(φ isa Real || φ isa Unitful.Quantity)
```

Comment:
Everything with units is a `Unitful.Quantity`, you might wanna check explicitly for something having no dimension:
```suggestion
    if !(φ isa Real || φ isa Unitful.Quantity{<:Real, NoDims})
```